server/api/user/user.controller.js

```diff
@@ -340,3 +347,43 @@ exports.removeTech = function(req,res){
         }
     });
 };
+
+/**
+ * Set reset token for user and email it the password token will expire after 24 hours.
+ */
+exports.resetPassword = function(req, res){
+    var userEmail = req.body.email;
+    User.findOne({
+        email: userEmail.toLowerCase()
+    }, function (err, user){
+        if (err) return res.json(401, err);
+        if (!user) return res.json(401, {message: 'User does not exist'});
+
+        crypto.randomBytes(12, function(ex, buf) {
+            var token = buf.toString('hex');
+            user.passwordResetToken = token;
+            
+            // Get tomorrow's date
+            var tomorrow = new Date();
+            tomorrow.setDate(tomorrow.getDate() + 1);
+            
+            user.passwordResetExpiration = tomorrow;
+
+            user.save(function(err){
+                if (err) return validationError(res,err);
+                res.send(200);
+
+                // email token to user
+                email.send(userEmail,
+                    "Observatory3 password reset",
+                    "Please go to the following url to reset your password\n\n\
+                    " + config.addr + "/login?token=" + user.passwordResetToken + "\n\n\
```

Comment:
not totally sure it would render properly- i've seen sites email just the link (some clients don't allow html?). Obviously something that should be looked at in the future